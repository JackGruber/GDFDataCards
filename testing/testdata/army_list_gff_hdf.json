{
    "id": "gIHRFheODbsP",
    "gameSystem": "gff",
    "armyId": "z65fgu0l29i4lnlu",
    "armyIds": [
        "z65fgu0l29i4lnlu"
    ],
    "armyFaction": null,
    "armyName": "Human Defense Force",
    "modified": "2024-10-27T14:13:06.459Z",
    "saveVersion": 3,
    "favourite": false,
    "armyVersions": [
        {
            "version": "3.4.1",
            "armyId": "z65fgu0l29i4lnlu"
        }
    ],
    "listPoints": 300,
    "list": {
        "id": "gIHRFheODbsP",
        "key": "jLeltQhTaedGSEamxRh4T",
        "name": "Human Defense Force",
        "description": "",
        "pointsLimit": 300,
        "campaignMode": false,
        "narrativeMode": false,
        "forceOrg": true,
        "gameSystem": "gff",
        "isCloud": false,
        "modified": "2024-10-27T14:13:06.459Z",
        "cloudModified": null,
        "activationCount": 9,
        "modelCount": 11,
        "units": [
            {
                "id": "95wirM-",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "Y2kwA",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "_nbz3zj",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "JpSZS",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "SG9cgFZ",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "86yAa",
                "selectedUpgrades": [
                    {
                        "instanceId": "_YQy7wwuT",
                        "upgradeId": "YePWZjo",
                        "optionId": "-YEWz"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "SScXOqX",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "W1Ate",
                "selectedUpgrades": [
                    {
                        "instanceId": "5ATNPJm0y",
                        "upgradeId": "Mj06h",
                        "optionId": "neXJq"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "SScXOqX",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "goMH9",
                "selectedUpgrades": [
                    {
                        "instanceId": "ErQXgBQJp",
                        "upgradeId": "2PGJ-qx",
                        "optionId": "1fW_LVD"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "Xuu-drz",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "KnFL4",
                "selectedUpgrades": [
                    {
                        "instanceId": "hO62GKrvA",
                        "upgradeId": "byolukJ",
                        "optionId": "Z8S0x-6"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "SScXOqX",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "djFq8",
                "selectedUpgrades": [
                    {
                        "instanceId": "JRC2w41El",
                        "upgradeId": "znb1L",
                        "optionId": "vn9Q1"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "3Tk4911",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "uCYmS",
                "selectedUpgrades": [
                    {
                        "instanceId": "SpZ-YGDGq",
                        "upgradeId": "wxW9CcN",
                        "optionId": "jubifhI"
                    },
                    {
                        "instanceId": "wPWCFIbpa",
                        "upgradeId": "F8Om0",
                        "optionId": "6omIv"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "Xuu-drz",
                "armyId": "z65fgu0l29i4lnlu",
                "selectionId": "lVd00",
                "selectedUpgrades": [
                    {
                        "instanceId": "hD86kTgPV",
                        "upgradeId": "An0_P",
                        "optionId": "fvlFb"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ]
    }
}