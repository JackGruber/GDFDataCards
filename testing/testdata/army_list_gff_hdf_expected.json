{
    "armyId": "z65fgu0l29i4lnlu",
    "gameSystemId": 3,
    "armyName": "Human Defense Force",
    "listPoints": 300,
    "listName": "300pt - HDF",
    "units": [
        {
            "type": "Storm Leader",
            "name": "Storm Leader",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "dwJg2Bu",
            "defense": 4,
            "quality": 4,
            "upgrades": [
                "B1",
                "C1",
                "cMlno"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Plasma Pistol",
                    "range": 12,
                    "specialRules": [],
                    "ap": "4"
                }
            ],
            "specialRules": [
                {
                    "key": "hero",
                    "name": "Hero",
                    "rating": "",
                    "label": "Hero"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "3",
                    "label": "Tough(3)"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                }
            ],
            "equipment": [
                {
                    "name": "Forward Observer",
                    "specialRules": [
                        "Take Aim"
                    ]
                }
            ]
        },
        {
            "type": "Sniper",
            "name": "Red Sniper",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "95wirM-",
            "defense": 5,
            "quality": 5,
            "upgrades": [],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Sniper Rifle",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "sniper",
                            "name": "Sniper",
                            "rating": "",
                            "modify": false,
                            "label": "Sniper"
                        }
                    ],
                    "ap": "1"
                }
            ],
            "specialRules": [
                {
                    "key": "scout",
                    "name": "Scout",
                    "rating": "",
                    "label": "Scout"
                },
                {
                    "key": "stealth",
                    "name": "Stealth",
                    "rating": "",
                    "label": "Stealth"
                }
            ]
        },
        {
            "type": "Special Weapon",
            "name": "Special Weapon",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "SG9cgFZ",
            "defense": 5,
            "quality": 5,
            "upgrades": [
                "G1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 6,
                    "count": 1,
                    "name": "Flamethrower",
                    "range": 12,
                    "specialRules": []
                }
            ],
            "specialRules": [
                {
                    "key": "relentless",
                    "name": "Relentless",
                    "rating": "",
                    "label": "Relentless"
                }
            ]
        },
        {
            "type": "Storm Trooper",
            "name": "Storm Trooper",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "Xuu-drz",
            "defense": 4,
            "quality": 4,
            "upgrades": [
                "B1",
                "D1",
                "I1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 3,
                    "count": 1,
                    "name": "Shred Rifle",
                    "range": 18,
                    "specialRules": [
                        {
                            "key": "rending",
                            "name": "Rending",
                            "type": "ArmyBookRule",
                            "label": "Rending",
                            "rating": ""
                        }
                    ]
                },
                {
                    "attacks": 2,
                    "count": 1,
                    "name": "Energy Fist",
                    "specialRules": [],
                    "ap": "4"
                }
            ],
            "specialRules": [
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                }
            ]
        },
        {
            "type": "Storm Trooper",
            "name": "Storm Trooper",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "Xuu-drz",
            "defense": 4,
            "quality": 4,
            "upgrades": [
                "B1",
                "D1",
                "I1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Plasma Rifle",
                    "range": 24,
                    "specialRules": [],
                    "ap": "4"
                }
            ],
            "specialRules": [
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                }
            ]
        },
        {
            "type": "Veteran",
            "name": "Veteran",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "SScXOqX",
            "defense": 5,
            "quality": 4,
            "upgrades": [
                "A1",
                "D1",
                "H1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Rifle",
                    "range": 24,
                    "specialRules": []
                }
            ],
            "specialRules": [],
            "equipment": [
                {
                    "name": "Demo Charges",
                    "specialRules": [
                        "AP(4) in melee"
                    ]
                }
            ]
        },
        {
            "type": "Veteran",
            "name": "Veteran",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "SScXOqX",
            "defense": 5,
            "quality": 4,
            "upgrades": [
                "A1",
                "D1",
                "H1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Rifle",
                    "range": 24,
                    "specialRules": []
                }
            ],
            "specialRules": []
        },
        {
            "type": "Special Weapon",
            "name": "Special Weapon",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "SG9cgFZ",
            "defense": 5,
            "quality": 5,
            "upgrades": [
                "G1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Grenade Launcher",
                    "range": 24,
                    "specialRules": [
                        {
                            "key": "blast",
                            "name": "Blast",
                            "type": "ArmyBookRule",
                            "label": "Blast(3)",
                            "modify": false,
                            "rating": "3"
                        }
                    ]
                }
            ],
            "specialRules": [
                {
                    "key": "relentless",
                    "name": "Relentless",
                    "rating": "",
                    "label": "Relentless"
                }
            ]
        },
        {
            "type": "Veteran",
            "name": "Veteran",
            "armyId": "z65fgu0l29i4lnlu",
            "id": "SScXOqX",
            "defense": 5,
            "quality": 4,
            "upgrades": [
                "A1",
                "D1",
                "H1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Rifle",
                    "range": 24,
                    "specialRules": []
                }
            ],
            "specialRules": []
        }
    ]
}